feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/presentation/staking/start/setupStakingType/SetupStakingTypeValidationFailureUi.kt

```diff
@@ -0,0 +1,47 @@
+package io.novafoundation.nova.feature_staking_impl.presentation.staking.start.setupStakingType
+
+import io.novafoundation.nova.common.base.TitleAndMessage
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.feature_staking_impl.R
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupStakingType.direct.EditingStakingTypeFailure
+import io.novafoundation.nova.feature_wallet_api.presentation.formatters.formatPlanks
+import io.novafoundation.nova.runtime.ext.StakingTypeGroup
+import io.novafoundation.nova.runtime.ext.group
+import io.novafoundation.nova.runtime.ext.isDirectStaking
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+
+fun handleSetupStakingTypeValidationFailure(chainAsset: Chain.Asset, error: EditingStakingTypeFailure, resourceManager: ResourceManager): TitleAndMessage {
+    return when (error) {
+        is EditingStakingTypeFailure.AmountIsLessThanMinStake -> {
+            val amount = error.minStake.formatPlanks(chainAsset)
+            when {
+                error.stakingType.isDirectStaking() -> TitleAndMessage(
+                    resourceManager.getString(R.string.setup_staking_type_staking_amount_is_less_than_min_amount_title),
+                    resourceManager.getString(R.string.setup_staking_type_direct_staking_amount_is_less_than_min_amount_message, amount)
+                )
+                else -> TitleAndMessage(
+                    resourceManager.getString(R.string.setup_staking_type_staking_amount_is_less_than_min_amount_title),
+                    resourceManager.getString(R.string.setup_staking_type_staking_amount_is_less_than_min_amount_fallback_message, amount)
+                )
+            }
+        }
+
+        is EditingStakingTypeFailure.StakingTypeIsAlreadyUsing -> {
+            return when (error.stakingType.group()) {
+                StakingTypeGroup.PARACHAIN,
+                StakingTypeGroup.RELAYCHAIN -> TitleAndMessage(
+                    resourceManager.getString(R.string.setup_staking_type_already_used_title),
+                    resourceManager.getString(R.string.setup_staking_type_direct_already_used_message)
+                )
+                StakingTypeGroup.NOMINATION_POOL -> TitleAndMessage(
+                    resourceManager.getString(R.string.setup_staking_type_already_used_title),
+                    resourceManager.getString(R.string.setup_staking_type_pool_already_used_message)
+                )
+                StakingTypeGroup.UNSUPPORTED -> TitleAndMessage(
+                    resourceManager.getString(R.string.setup_staking_type_already_used_title),
+                    resourceManager.getString(R.string.setup_staking_type_unsupported_staking_type_used_fallback_message)
+                )
+            }
+        }
```

Comment:
Please avoid switching over `StakingTypeGroup` when its unesseary. We should not hardcode `Direct staking` or `Pool staking` labels into strings since we already have them separately and can use right one as a formatting argument. We can format staking type with `resourceManager.formatStakingTypeLabel()`

The less places we have switching over `StakingTypeGroup` at, the easier it will be to add a new StakingType support

Reply:
I think it's possible for the first condition but for the last we use an inappropriate resources
<img width="375" alt="image" src="https://github.com/novasamatech/nova-wallet-android/assets/107959809/61ea472f-99ab-4aa5-94b3-8479ddc33e43">

Reply:
But we already have a method `resourceManager.formatStakingTypeLabel()` that performs this switching so no need to dublicate this logic twice